tests/dbos.test.ts

```diff
@@ -275,8 +275,80 @@ describe('dbos-tests', () => {
       );
     });
   });
+
+  test('workflow-withWorkflowTimeout', async () => {
+    const workflowID: string = randomUUID();
+    await DBOS.withNextWorkflowID(workflowID, async () => {
+      await DBOS.withWorkflowTimeout(100, async () => {
+        await expect(DBOSTimeoutTestClass.blockedWorkflow()).rejects.toThrow(
+          new DBOSWorkflowCancelledError(workflowID),
+        );
+      });
+    });
+  });
+
+  test('workflow-timeout-startWorkflow-params', async () => {
+    const workflowID = randomUUID();
+    const handle = await DBOS.startWorkflow(DBOSTimeoutTestClass, { workflowID, timeout: 100 }).blockedWorkflow();
+    await expect(handle.getResult()).rejects.toThrow(new DBOSWorkflowCancelledError(workflowID));
+  });
+
+  test('parent-workflow-withWorkflowTimeout', async () => {
+    const workflowID: string = randomUUID();
+    await DBOS.withNextWorkflowID(workflowID, async () => {
+      await DBOS.withWorkflowTimeout(100, async () => {
+        await expect(DBOSTimeoutTestClass.blockingParentStartWF()).rejects.toThrow(
+          new DBOSWorkflowCancelledError(workflowID),
+        );
+      });
+    });
+  });
+
+  test('parent-workflow-timeout-startWorkflow-params', async () => {
+    const workflowID = randomUUID();
+    const handle = await DBOS.startWorkflow(DBOSTimeoutTestClass, { workflowID, timeout: 100 }).blockingParentStartWF();
+    await expect(handle.getResult()).rejects.toThrow(new DBOSWorkflowCancelledError(workflowID));
+  });
+
+  test('direct-parent-workflow-withWorkflowTimeout', async () => {
+    const workflowID: string = randomUUID();
+    await DBOS.withNextWorkflowID(workflowID, async () => {
+      await DBOS.withWorkflowTimeout(100, async () => {
+        await expect(DBOSTimeoutTestClass.blockingParentDirect()).rejects.toThrow(
+          new DBOSWorkflowCancelledError(workflowID),
```

Comment:
Can we also check the workflow's status is set to CANCELLED? I'm not sure if it is.

Reply:
Can do